VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +123,90 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
     public void onClickViewComments(@NonNull DiscussionComment comment) {
         router.showCourseDiscussionComments(getActivity(), comment, discussionThread.isClosed());
     }
+
+    private class CommentsLoader {
+
+        private InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback;
+
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private ThreadComments unendorsedComments;
+        private int nextPage = 1;
+        private boolean isQuestionTypeThread = false;
+        private boolean isEndorsedFetched = false;
+
+        public CommentsLoader(boolean isQuestionTypeThread) {
```

Comment:
I guess this is meant to be a standalone class, which is why the `isQuestionTypeThread` parameter is defined here. However, since we're already referencing the `DiscussionThread` when using it's ID, we might as well have that passed here and query it's type when needed, instead of defining a redundant variable. Then if we also take a `Context` as parameter, this class can be defined as `static` and standalone.